Source/Urho3D/CSharp/Managed/Core/Context.cs

```diff
@@ -146,6 +152,48 @@ internal HandleRef CreateObject(uint managedType)
             return (T) GetSubsystem(new StringHash(typeof(T).Name));
         }
 
+        public ConfiguredTaskAwaitable<bool> ToMainThreadAsync()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            InvokeOnMainThread(() => tcs.TrySetResult(true));
+            return tcs.Task.ConfigureAwait(false);
+        }
+
+        public void InvokeOnMainThread(Action func)
```

Comment:
There is `WorkQueue::PostTaskForMainThread`, i believe we should use it for this purpose